test/system/335-corrupt-layers.bats

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# All tests in here perform nasty manipulations on image storage.
+#
+
+load helpers
+
+###############################################################################
+# BEGIN setup/teardown
+
+# Create a scratch directory; this is what we'll use for image store and cache
+if [ -z "${PODMAN_CORRUPT_TEST_WORKDIR}" ]; then
+    export PODMAN_CORRUPT_TEST_WORKDIR=$(mktemp -d --tmpdir=${BATS_TMPDIR:-${TMPDIR:-/tmp}} podman_corrupt_test.XXXXXX)
+fi
+
+# All tests in this file (and ONLY in this file) run with a custom rootdir
+function setup() {
+    skip_if_remote "none of these tests run under podman-remote"
+    _PODMAN_TEST_OPTS="--storage-driver=vfs --root ${PODMAN_CORRUPT_TEST_WORKDIR}/root"
+}
+
+function teardown() {
+    # No other tests should ever run with this custom rootdir
+    unset _PODMAN_TEST_OPTS
+
+    is_remote && return
+
+    # Clean up
+    umount ${PODMAN_CORRUPT_TEST_WORKDIR}/root/overlay || true
+    if is_rootless; then
+        run_podman unshare rm -rf ${PODMAN_CORRUPT_TEST_WORKDIR}/root
+    else
+        rm -rf ${PODMAN_CORRUPT_TEST_WORKDIR}/root
+    fi
+}
+
+# END   setup/teardown
+###############################################################################
+# BEGIN primary test helper
+
+# This is our main action, invoked by every actual test. It:
+#    - creates a new empty rootdir
+#    - populates it with our crafted test image
+#    - removes [ manifest, blob ]
+#    - confirms that "podman images" throws an error
+#    - runs the specified command (rmi -a -f, prune, reset, etc)
+#    - confirms that it succeeds, and also emits expected warnings
+function _corrupt_image_test() {
+    run_podman info --format "{{.Store.GraphRoot}}"
+    indexPath=$output/*-layers/layers.json
+
+    run_podman pull $IMAGE
+    run_podman image exists $IMAGE
+
+    run jq 'del(.[0])' $indexPath > $indexPath
```

Comment:
/hold

I don't think this does what you think it does. I need time to look into it.

Reply:
Good catch.  I see it now.  Writing back like this will empty the entire file.  It yields effectively the same error but is indeed not what I inteded.

Reply:
@edsantiago how about:
``bash
run jq 'del(.[0])' $indexPath > $indexPath.tmp
run mv $indexPath.tmp $indexPath
```

Reply:
Closer, but still won't work: `run` is evil, it's a super-misdirecting problem with BATS. You should almost never use `run`. So, yes, do the above but remove all `run`.

Reply:
...except that the `jq` is malformed, so that will fail too.

Reply:
```diff
diff --git a/test/system/335-corrupt-layers.bats b/test/system/335-corrupt-layers.bats
index 54eee9992..3bd67de8d 100644
--- a/test/system/335-corrupt-layers.bats
+++ b/test/system/335-corrupt-layers.bats
@@ -52,7 +52,9 @@ function _corrupt_image_test() {
     run_podman pull $IMAGE
     run_podman image exists $IMAGE

-    run jq 'del(.[0])' $indexPath > $indexPath
+    indexTmp="$(realpath $indexPath).tmp.$$"
+    jq 'del(.[0])' $indexPath > $indexTmp
+    mv -f $indexTmp $indexPath
     # A corrupted image won't be marked as existing.
     run_podman 1 image exists $IMAGE
```

Reply:
Done.  As always, thanks for you guidance, @edsantiago!